pkg/k8s/api/v1alpha1/edgedevice_types.go

```diff
@@ -23,6 +23,13 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// TCPSetting defines TCP forward settings
+type TCPSetting struct {
+	// TCPServerAddress *string `json:"TCPServerAddress,omitempty"`
+	NetworkType *string `json:"NetworkType,omitempty"`
```

Comment:
we can add the validation here using `https://book.kubebuilder.io/reference/markers/crd-validation.html`